src/main/kotlin/com/terraformation/backend/customer/AppNotificationService.kt

```diff
@@ -337,6 +342,26 @@ class AppNotificationService(
     }
   }
 
+  @EventListener
+  fun on(event: ModuleEventStartingEvent) {
+    systemUser.run {
+      val moduleEvent = moduleEventStore.fetchOneById(event.eventId)
+      val renderMessage = {
+        messages.moduleEventStartingNotification(
+            parentStore.getEventType(event.eventId)!!,
+            ModuleEventNotifier.notificationLeadTime,
+            parentStore.getModuleName(event.eventId)!!)
```

Comment:
How about having the module ID in `EventModel` (which seems like it'd be generally useful anyway) and then fetching the module from the module store to get its name?

My concern here is mostly that `ParentStore` could easily turn into a grab-bag of "fetch this random field on this random table" functions. Which isn't the end of the world, but the reason it exists is so the other `Store` classes can navigate relationships without having to have each other as dependencies, which can lead to dependency cycles. Service classes like this one are free to depend on store classes as needed, no concerns about cycles.

We probably ought to document more of this kind of architectural thinking, admittedly; there's no great way to discover any of it right now.